src/v/utils/fragmented_vector.h

```diff
@@ -222,6 +222,57 @@ class fragmented_vector {
         _capacity = 0;
     }
 
+    /**
+     * Return the number of elements the vector has allocated space for.
+     */
+    size_t capacity() const { return _capacity; }
+
+    /**
+     * Resizes the vector to contain \p count elements. If the current size is
+     * less than \p count then default-initialized elements are added. If the
+     * current size is larger then elements are removed.
+     *
+     * The expected use case for this is to allocate a large vector in a fiber
+     * using a series of smaller resize() invocations allowing for cooperative
+     * yield calls to be inserted to avoid reactor stalls. The optimal strategy
+     * would be to fill the latest fragment by inspecting the current capacity()
+     * and then to make allocations in increments of elements_per_fragment().
+     */
+    void resize(const size_t count) {
+        if (_size == count) {
+            return;
+        }
+
+        if (_size > count) {
+            pop_back_n(size() - count);
+            return;
+        }
+
+        auto deficit = count - _size;
+        while (deficit) {
+            if (_size == _capacity) {
+                std::vector<T> frag;
+                if (deficit >= elems_per_frag) {
+                    frag.resize(elems_per_frag);
+                } else {
+                    frag.reserve(elems_per_frag);
+                    frag.resize(deficit); // preserves capacity
+                }
+                const auto add_count = frag.size();
+                _frags.push_back(std::move(frag));
+                _capacity += elems_per_frag;
+                _size += add_count;
+                deficit -= add_count;
+                continue;
+            }
+
+            const auto add_count = std::min(deficit, _capacity - _size);
+            _frags.back().resize(_frags.back().size() + add_count);
```

Comment:
nit: could resize to `elems_per_frag` and skip the arithmetic

Reply:
doesn't that only work if `deficit` is larger than the remaining space in the fragment? what if the fragment was half full but `deficit` was `1`?

Reply:
Yeah, you're right. Makes sense